Interface Class Specifications/Display/InterfaceSpec-Display.md

```diff
@@ -0,0 +1,86 @@
+
+## Messages
+In general, to work smoothly with the rest of the OSVR system, distance units are meters and velocity units are meters/second.
+
+### Display
+#### Data
+- type (enum in fixed|hmd)
+- vendor (string)
+- model: (string)
+- product URI (uri-string RFC 3986)
+- driver URI (uri-string RFC 3986)
+- notes: string
+- number of DisplayInputs (whole number)
+- number of logical DisplaySurfaces (whole number)
+- k1[3] rgb radial distortion supported (boolean, hmd only?) and priority (whole number)
+- RGB radial distortion supported (boolean, hmd only?, supercedes k1[3] in most cases) and priority (whole number)
+- mono point mesh distortion supported (boolean, hmd only?) and priority (whole number)
+- RGB mesh distortion supported (boolean, hmd only?) and priority (whole number)
+
+#### Rationale
+The Display message is the root object of the Display interface. It references both the physical DisplayInputs and logical DisplaySurfaces, which define the mapping between them.
+
+### DisplayInput
+> one per number of display inputs in Display message.
+
+#### Data
+- resolution (width, height in whole numbers)
+- vendor PNPID (string)
+- product id (string)
+- platform-specific hardware instance path to uniquely identify it (??? string?)
+- if part of the desktop, origin on the desktop (X and Y in whole numbers)
+- On Windows, dx adapter number (??? string?)
+- logical id or platform-agnostic identifier (string? or int? referenced by logical topology)
+- location and orientation (physical orientation and location? relative to what?)
+- eye relief/lense-to-screen distance (if not represented with location/orientation above)
+- Scan-out direction (unit OSVR_Vec2)
+- Scan-out framerate (analog/float)
+- estimated latency to start of render in ms (analog/float)
+
+#### Rationale
+The DisplayInput message represents a physical display input.
+
+### DisplaySurface
+> one per number of logical display surfaces in Display message
+
+#### Data
+- logical id of display input for this surface (from DisplayInput message)
+- eye index (whole number)
+- viewer index (whole number)
+- viewport (relative to this surface's corresponding DisplayInput)
+- rotation (enum in landscape|portrait|landscape-flipped|portrait-flipped in clockwise order, relative to corresponding DisplayInput)
+- lens center of projection (OSVR_Vec2, HMD only?)
+- hidden area mesh (vector of OSVR_Vec2, HMD only?, optional)
+- RGB radial distortion center of projection (OSVR_Vec2, HMD only? optional)
+- mono point distortion mesh (HMD only?, optional)
+- RGB distortion mesh (HMD only?, optional)
+
+#### Rationale
+This is one serialization of the logical viewer-eye-surface mapping to physical display input hardware. It assumes one DisplayInput per DisplaySurface, and one combination of (viewer, eye) per surface. If this is not the case, we'll have to serialize these differently.
```

Comment:
I think the one-DisplayInput-per-DisplaySurface assumption might break in the case of some CAVE configurations. Think of shooting multiple projectors onto a single screen. (I don't know if that sort of configuration is within the scope of this proposal or even within the scope of OSVR, but it might be something worth considering.)